cli/js/lib.deno.ns.d.ts

```diff
@@ -1509,47 +1511,48 @@ declare namespace Deno {
    * `Uint8Array`.
    *
    * Resolves to the number of bytes written and the remote address. */
-  export function recvfrom(rid: number, p: Uint8Array): Promise<[number, Addr]>;
+  //   export function recvfrom(rid: number, p: Uint8Array): Promise<[number, Addr]>;
 
   /** **UNSTABLE**: new API, yet to be vetted.
    *
    * A generic transport listener for message-oriented protocols. */
-  export interface UDPConn extends AsyncIterator<[Uint8Array, Addr]> {
+  export interface DatagramConn<T> extends AsyncIterator<[Uint8Array, T]> {
     /** **UNSTABLE**: new API, yet to be vetted.
      *
      * Waits for and resolves to the next message to the `UDPConn`. */
-    receive(p?: Uint8Array): Promise<[Uint8Array, Addr]>;
+    receive(p?: Uint8Array): Promise<[Uint8Array, T]>;
     /** UNSTABLE: new API, yet to be vetted.
      *
      * Sends a message to the target. */
-    send(p: Uint8Array, addr: UDPAddr): Promise<void>;
+    send(p: Uint8Array, addr: T): Promise<void>;
     /** UNSTABLE: new API, yet to be vetted.
      *
      * Close closes the socket. Any pending message promises will be rejected
      * with errors. */
     close(): void;
     /** Return the address of the `UDPConn`. */
-    readonly addr: Addr;
-    [Symbol.asyncIterator](): AsyncIterator<[Uint8Array, Addr]>;
+    readonly addr: T;
+    [Symbol.asyncIterator](): AsyncIterator<[Uint8Array, T]>;
   }
 
   /** A generic network listener for stream-oriented protocols. */
-  export interface Listener extends AsyncIterator<Conn> {
+  export interface Listener<T> extends AsyncIterator<Conn> {
```

Comment:
Better to make this not generic.

Reply:
Should I change to an interface union as suggested for the Conn interface?
IMO it should either be Generic or have a simple Addr interface like Go has for all its connections. This way the user would not have to conditionally branch in order to get the address properties.

Reply:
I’d prefer to go the Go route.. but I need to discuss this with Bert. We want to avoid having generics for relatively minor differences in interfaces because genetics tend to propagate outward to user interfaces making everything more complex...

Reply:
I'd vouch for the Go route, it feels like the best alternative. After implementing the Generic Interfaces, I found that the outward propagation implied by this implementation could be just as bad for the end Deno user, as the conditional branching that is necessary with an interface union approach. 
I'll wait for your discussion with @piscisaureus and implement what you guys decide on :)